airflow/models/dagrun.py

```diff
@@ -728,9 +728,12 @@ def _filter_tis_and_exclude_removed(dag: "DAG", tis: List[TI]) -> Iterable[TI]:
                 try:
                     ti.task = dag.get_task(ti.task_id)
                 except TaskNotFound:
-                    self.log.error("Failed to get task for ti %s. Marking it as removed.", ti)
-                    ti.state = State.REMOVED
-                    session.flush()
+                    if ti.state == State.REMOVED:
+                        pass  # ti has already been removed, just ignore it
+                    else:
```

Comment:
```suggestion
                    if ti.state != State.REMOVED:
```

nit: we can probably just go with this.

Reply:
Yes that seems fair. I think that is also explicit enough indeed.